### ISSUE TITLE
Results_ucr_resamples

### ISSUE THREAD

**Issue by peter943** [angus924/rocket#6]
Hi,
I have learnt your paper these day, which is very enlightening to me. I have a question about Results_ucr_resamples. I do not how to reproduce the results and what does resample mean. Could you please give me some help?

**Comment by angus924** [comment]
Great question.

Resampling is where we 'shuffle' the original dataset to create new training / test splits, similar to [`StratifiedShuffleSplit`](https://scikit-learn.org/stable/modules/generated/sklearn.model_selection.StratifiedShuffleSplit.html).

For comparability with other published results (namely, http://timeseriesclassification.com/results.php), we use [`resampleTrainAndTestInstances(...)`](https://github.com/uea-machine-learning/tsml/blob/558ae0d13e32f5563ea58207bd676cfaf5ded0dd/src/main/java/utilities/InstanceTools.java#L153) to recreate the same resamples as in [Bagnall et al. (2017)](https://doi.org/10.1007/s10618-016-0483-9).  The results in [`results_ucr_resamples.csv`](https://github.com/angus924/rocket/blob/master/results/results_ucr_resamples.csv) correspond to the first 10 resamples (*not* including the original training / test split).

**Comment by peter943** [comment]
@angus924 Thanks for your help. The results in Results_ucr_resamples.csv are the average of the first 10 resamples. This understanding is right?

**Comment by angus924** [comment]
That's correct.

**Comment by peter943** [comment]
@angus924 Ok，thanks a lot.

**Comment by peter943** [comment]
@angus924
Hi，I have another question about 'development' and 'holdout' dataset.

In your paper, you say “Accordingly, we chose to develop the method on a subset of 40 randomly-selected datasets from the 85 ‘bake off’ datasets. We refer to these as the ‘development’ datasets. We provide a separate evaluation of the performance of Rocket on the ‘development’ datasets and the remaining ‘holdout’ datasets in Appendix B.” 

I do not understand what role do these two data sets play. Maybe  'development'  dataset is used to adjust hyperparameters and testing, and  'holdout' dataset is just for testing. 

Could you please give me some help? Thansks a lot.

**Comment by angus924** [comment]
Another great question.

> Maybe 'development' dataset is used to adjust hyperparameters and testing, and 'holdout' dataset is just for testing.

Essentially, yes.

In order to make basic choices about, e.g., kernel length, dilation, etc., we needed some datasets to work with.  We randomly selected 40 'development' datasets (from the 85 'bake off' datasets) for this purpose, in order to avoid overfitting the whole archive.  (None of the remaining 45 'holdout' datasets, or the 43 additional datasets added to the archive in 2018, were used.)

As shown in the sensitivity analysis, two parameters matter a lot more than anything else: dilation and pooling (ppv).

Ultimately, accuracy (vs other SOTA methods) is more or less the same for the 'development', 'holdout', and additional datasets, as well as for resamples of all of these datasets.

**Comment by peter943** [comment]
@angus924, I understand, thank you!
